ethereum/rlp/src/main/java/org/hyperledger/besu/ethereum/rlp/RLPInput.java

```diff
@@ -221,6 +221,11 @@ public interface RLPInput {
    *     long.
    */
   default int readUnsignedByte() {
+    if (isZeroLengthString()) {
+      // Decode an empty string (0x80) as an unsigned byte with value 0
+      readBytes();
+      return 0;
+    }
```

Comment:
I suspect we should be fixing the serialisation of 0 as well to ensure we write it as an 0 length string. Otherwise we won't be to spec.

Reply:
I've created an issue to look at encoding and decoding of 0. This PR fixes the decoding of the Nethermind hello, which was out priority.